### ISSUE TITLE
Integrate SPIFFS example - No error but no data read either 

### ISSUE THREAD

**Issue by ClamentJohn** [pellepl/spiffs#190]
No errors received.
I've followed the Integrate SPIFFS [tutorial](https://github.com/pellepl/spiffs/wiki/Integrate-spiffs) to use it with my NOR flash (AT45DB041), I use an STM32F103.  But when I run the example it doesn't read anything or write anything.
I've tried printing every data that reaches my driver and they are not strings so I receive a bunch of '?'.
I wrote the driver for this and I assume this works properly (I tried to write, read and chip erase) 

AT45DB - 4Mbit data flash, one bock is 2KB
Configuration
```
cfg.phys_size = 512*1024; // use all spi flash
  cfg.phys_addr = 0; // start spiffs at start of spi flash
  cfg.phys_erase_block = 4*1024; // according to datasheet
  cfg.log_block_size = 4*1024; // let us not complicate things
  cfg.log_page_size = LOG_PAGE_SIZE; // as we said
```

The output I receive
```
errno after write  = 0
errno after close = 0
errno after read  = 0
-->  <--
```
What could I be missing?

**Comment by ClamentJohn** [comment]
I hadn't set `SPIFFS_USE_MAGIC` now that I've set it I'm unable to mount it. I followed the config from the wiki and have config'ed it according to my NOR flash. And for formating I've mount, unmount, format, mount. But still, no file system is seen in the flash.

**Comment by pellepl** [comment]
Hi John,

We're in the middle of a heavily delayed deadline at work and I have no
time right now I'm afraid.
However, everything you mention indicates that your spiflash functions are
not working. My best bet would be to verify that you actually can
read/write/erase the spiflash.

Cheers/ P


Den 15 dec. 2017 13:08 skrev "Clament John" <notifications@github.com>:

I hadn't set SPIFFS_USE_MAGIC now that I've set it I'm unable to mount it.
I followed the config from the wiki and have config'ed it according to my
NOR flash. And for formating I've mount, unmount, format, mount. But still,
no file system is seen in the flash.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/pellepl/spiffs/issues/190#issuecomment-351990606>, or mute
the thread
